src/test/fuzz/utxo_total_supply.cpp

```diff
@@ -0,0 +1,158 @@
+        LOCK(cs_main);
+        ::ChainstateActive().ForceFlushStateToDisk();
+        assert(GetUTXOStats(&ChainstateActive().CoinsDB(), utxo_stats));
+        // Check that miner can't print more money than they are allowed to
+        assert(circulation == utxo_stats.nTotalAmount);
+    };
+
+
+    // Update internal state to chain tip
+    StoreLastTxo();
+    UpdateUtxoStats();
+    assert(ChainActive().Height() == 0);
+    // Get at which height we duplicate the coinbase
+    int64_t duplicate_coinbase_height;
+    CScript duplicate_coinbase_script;
+    {
+        // A chance of 1 in 20 for the duplicate coinbase to be within the first 100 blocks feels about right
+        duplicate_coinbase_height = fuzzed_data_provider.ConsumeIntegralInRange(0, 20 * COINBASE_MATURITY);
+        // Mine the first block with this duplicate
+        current_block = PrepareNextBlock();
+        StoreLastTxo();
+
+        // Create duplicate (CScript should match exact format as in CreateNewBlock)
+        CMutableTransaction tx{*current_block->vtx.back()};
+        duplicate_coinbase_script = CScript() << duplicate_coinbase_height << OP_0;
+        tx.vin.at(0).scriptSig = duplicate_coinbase_script;
+
+        // Mine block and create next block template
+        current_block->vtx.back() = MakeTransactionRef(tx);
+        current_block->hashMerkleRoot = BlockMerkleRoot(*current_block);
+        assert(!MineBlock(current_block).IsNull());
+        circulation += GetBlockSubsidy(ChainActive().Height(), Params().GetConsensus());
+
+        assert(ChainActive().Height() == 1);
+        UpdateUtxoStats();
+        current_block = PrepareNextBlock();
+        StoreLastTxo();
+    }
+
+    while (fuzzed_data_provider.remaining_bytes()) {
+        const auto action = static_cast<Action>(fuzzed_data_provider.ConsumeIntegralInRange(0, 2));
+        switch (action) {
+        case Action::CREATE_INPUT: {
+            CMutableTransaction tx{*current_block->vtx.back()};
+            AppendRandomTxo(tx);
+            current_block->vtx.back() = MakeTransactionRef(tx);
+            StoreLastTxo();
+            break;
+        }
+        case Action::CREATE_TX: {
+            CMutableTransaction tx{};
+            AppendRandomTxo(tx);
+            current_block->vtx.push_back(MakeTransactionRef(tx));
+            StoreLastTxo();
+            break;
+        }
+        case Action::CREATE_BLOCK: {
+            ReGenerateCommitments(*current_block);
+            const bool was_valid = !MineBlock(current_block).IsNull();
+
+            const auto prev_utxo_stats = utxo_stats;
+            if (was_valid) {
+                circulation += GetBlockSubsidy(ChainActive().Height(), Params().GetConsensus());
+
+                if (duplicate_coinbase_height == ChainActive().Height()) {
+                    // we mined the duplicate coinbase
+                    assert(current_block->vtx.at(0)->vin.at(0).scriptSig == duplicate_coinbase_script);
```

Comment:
use .front() instead of .at(0)

Reply:
why?

.at() asserts that the element exists, which I like